File: medicalcheckrest/WMedicalCheck/WMedicalCheck/medicalcheck/src/main/java/ch/hearc/medicalcheck/service/FollowService.java
package ch.hearc.medicalcheck.service;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import ch.hearc.medicalcheck.model.Follow;
import ch.hearc.medicalcheck.repository.FollowRepository;

/*
* Project   : Medical Check Rest
* Authors   : William Bikuta, Milán Cerviño, Ilyas Boillat, David Oktay
* Date      : 28.01.2022
* Class     : INF3dlm-a
* */

/**
 * bridge between repository and controller
 * each methods is explained in the associate controller
 * check them for more information
 */

@Service
public class FollowService {
	@Autowired
	private FollowRepository followRepository;

	public List<Follow> getAll() {
		return followRepository.findAll();
	}
	
	public List<Follow> getAllFollowByIdUserCareKeeper(Integer idUserCareKeeper) {
		return followRepository.getAllFollowByIdUserCareKeeper(idUserCareKeeper);
	}
	
	public List<Follow> getAllFollowByIdUserPatient(Integer idUserPatient) {
		return followRepository.getAllFollowByIdUserPatient(idUserPatient);
	}


	public Follow get(Integer id) {
		return followRepository.findById(id).get();
	}
	
	public Follow create(Follow follow) {
		return followRepository.save(follow);
	}

	public Follow update(Integer idFollow, Follow follow) {
		Optional<Follow> followData = followRepository.findById(follow.getId());

		if (followData.isPresent()) {
			follow.setId(idFollow);
			return followRepository.save(follow);
		}
		return null; 
	}

	public void delete(Integer id) {
		followRepository.deleteById(id);
	}
}

File: medicalcheck/app/src/main/java/ch/hearc/medicalcheck/services/FollowService.kt
package ch.hearc.medicalcheck.services

import ch.hearc.medicalcheck.models.Follow
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.*

/*
* Project   : Medical Check Mobile
* Authors   : William Bikuta, Milán Cerviño, Ilyas Boillat, David Oktay
* Date      : 28.01.2022
* Class     : INF3dlm-a
* */

interface  FollowService {
    @GET("follows")
    fun getAll(): Call<List<Follow>>

    @GET("follows/{id}")
    fun get(@Path("id") id : Int): Call<Follow>

    @GET("follows/usercarekeeper/{idUserCareKeeper}")
    fun getAllFollowByIdUserCareKeeper(@Path("idUserCareKeeper") idUserCareKeeper: Int): Call<List<Follow>>

    @POST("follows")
    fun create(@Body follow: Follow): Call<Follow>

    @DELETE("follows/{id}")
    fun delete(@Path("id") id: Int): Call<ResponseBody>

    @PUT("follows/{id}")
    fun update(@Path("id") id : Int, @Body follow: Follow): Call<Follow>
}